### DISCUSSION TITLE
filament modelling enhancement (ie for Pressure Advance)

### DISCUSSION THREAD

**Discussion by CooperGerman** [OrcaSlicer/OrcaSlicer#4574] [Ideas]
Hi,
just wanted to discuss about filament attributes such as for instance the Pressure Advance.

As it stands orcaslicer associates a PA value to a filament with no further tweaking possible.
I found that i had multiple printers, different extruders, different nozzles etc ... and that the PA
value i had calibrated for one filament wasd only true for the current printer/extruder/nozzle combo.

I 'm not sure if I'm not understanding this correctly but it seems like i would want to associated a PA value in the filament configuration windows for a specific machine (nozzle size variations for a machine seems to be others machine objects).

This also might be true for temperature settings as i found that .15 or .25 mm nozzles on the same printer need different temperatures than  the standard .4mm one ...

Thanks for any insight

**Comment by TexZeTech** [comment]
They way I was setting up my printer config

- Printer section
 < Name >< nozzle >
eg; Voron Trident 350 0.4 nozzle

- Filament
 < Short printer name > < filament brand placeholder or generic base> < material type > < colour >
eg; Voron Generic ABS 
eg; Voron Prusament PETG
eg; Voron Prusament PETG Ultramarine Blue Transparent

- Process section
< Layer height >< general quality name >< limited to printer name >
eg; 0.20mm Standard @ Voron

Its a lot of manually editing the config as you can say if using nozzle type X limit to filaments xyz. You can then load assorted values but if you are running Klipper I would recommend getting [Spoolman](https://github.com/Donkie/Spoolman#integration-status) up and get macros associated on your printer. You should be able to do a similar setup in OrcaSlicer if you are using other printer though just using gcode.

I know there is talk of better integration with Spoolman as well.

**Reply by CooperGerman** [reply]
Thanks for your reply, i understand what you are going for. This is indeed very similar to the way i did it too.
I forked moonraker to modify spoolman in order to support my needs (see : https://github.com/CooperGerman/moonraker_spoolman_enhancement)

I'm not that happy with the way some information is stored on orcaslicer side and some (i've chosen to store in spoolman db in a non intended way).

This discussion echoes with another i'm having on klipper discourse (https://klipper.discourse.group/t/way-to-post-pre-process-gcode-through-moonraker/15058/17).

Ideally i would manage my filament db through orcaslicer and have only one filament object that can be used by multiple printers and they all fetch their motion-dependent values and temperature values from the filament. If no such way of organizing data is intended to be implement in orca it seems like my only other option is to keep on with my patchy solution involving storing non intended information in the spoolman db.

Thanks for sharing your experience, i'll keep you posted for any advancement i'll make regarding this topic !

**Reply by TexZeTech** [reply]
I agree it would be amazing to reference all of that right off the SpoolMan DB instead of storing locally. 

This is the PR https://github.com/SoftFever/OrcaSlicer/pull/4771

I say its a step in the correct direction.

**Reply by CooperGerman** [reply]
Wow thx i did not see this one before, i'm going to try this one out for sure !

**Comment by Druinz** [comment]
+1
The setting grouping doesn't seem logical and in my opinion separates a lot of related settings. I think it would easier to have all settings (printer, filament, nozzle) grouped so we can just create a bunch of unique profiles instead of playing mix-n-match and trying to keep the settings matched appropriately. 
That may just be my mentality coming from Cura, but having to dig through several 'unrelated' menus to get to print settings that I need to change based off of one setting seems counter-intuitive..
IE the same model on the same printer, just with a larger nozzle & layer height calls for:
1. Change my nozzle width setting
2. Find the wall loops setting under my 'process' preset that is picked from the nozzle preset and update that
3. Due to changing my layer height so drastically, I now need to change 'filament' settings to adjust PA, even though I'm using the same filament. So this now branches off two 'filament' profiles for the same filament that I then have to match to my 'process' and 'nozzle' settings, rather than just grouping it accordingly.

These are all changes cascaded from one change, the nozzle and following that, layer height. But I had to change process settings and 'filament' settings. It just seems over-engineered to split these, when the simpler solution would work better (at least for my workflow).

If someone has a better way of managing all these settings **centrally** with a profile, please let me know. I'm used to having to change a lot of settings per-print, but having to jump through different menus to find them is just a PITA frankly, since they're seemingly organized arbitrarily, so I find myself having to go through the different menus to find most settings until I just memorize where it lives.
